build_tools/cmake/iree_run_module_test.cmake

```diff
@@ -0,0 +1,276 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# Utility function to check expected failure for different platforms.
+function(platform_xfail OUTPUT_XFAIL XFAIL_ARGS)
+  set(${OUTPUT_XFAIL} FALSE PARENT_SCOPE)
+  if(XFAIL_ARGS)
+    if((CMAKE_SYSTEM_PROCESSOR STREQUAL "riscv64" AND
+        CMAKE_SYSTEM_NAME STREQUAL "Linux" AND
+        XFAIL_ARGS MATCHES "rv64") OR
+       (CMAKE_SYSTEM_PROCESSOR STREQUAL "riscv32" AND
+        CMAKE_SYSTEM_NAME STREQUAL "Linux" AND
+        XFAIL_ARGS MATCHES "rv32") OR
+       (ANDROID AND XFAIL_ARGS MATCHES "arm64-v8a") OR
```

Comment:
`ANDROID` does not imply arm64-v8a, necessarily. Can we make the check more precise here?

Reply:
https://android.googlesource.com/platform/ndk/+/refs/heads/master/build/cmake/android.toolchain.cmake#194 Android ndk which we used to build for arm64-v8a processor only has `set(CMAKE_SYSTEM_NAME Android)`. `ANDROID` variable is a shorthand for it https://cmake.org/cmake/help/latest/variable/ANDROID.html

Reply:
ahh.. I will add `CMAKE_ANDROID_ARCH_ABI` check of `arm64-v8a` in the function
https://android.googlesource.com/platform/ndk/+/refs/heads/master/build/cmake/android.toolchain.cmake#116

Reply:
Add the logic in `get_platform` function.